packages/lib-classifier/src/plugins/drawingTools/models/tools/Tool/Tool.js

```diff
@@ -107,9 +103,7 @@ const Tool = types
       // then this can be moved to the tools that should delete on invalid mark
       // transcription line, ellipse
       self.marks.forEach((mark) => {
-        if (!mark.isValid) {
-          self.deleteMark(mark)
-        }
+        mark.finish()
```

Comment:
OK, the "if mark is invalid, delete it" logic has been moved to mark.finish(). 👌 (pt 2/2)